### ISSUE TITLE
Add a delay after inputting the PIN to prevent PIN GUI lag

### ISSUE THREAD

**Issue by dabura667** [voisine/breadwallet#172]
I have been consistently having breadwallet freeze for the 1 second or so while working with my PIN number pad having my 4th digit in the "pressed" state (where it is darker than the rest)

I am on iPhone 5 with the latest iOS (the one that came with the Apple Watch app)

It's a small thing... but anything increasing the amount of time I need to worry about people looking over my shoulder is not a good thing.

Maybe delaying for like 1 ms might give the GUI time to return the PIN pad to the unpressed state?

**Comment by voisine** [comment]
I haven't been able to reproduce the lag you mention on a 6th gen iPod touch running iOS8.2, or an iPhone 4.

I'm curious what might be causing it. Did it only start after updating to 8.2?

**Comment by dabura667** [comment]
yes. It is rather noticeable for me... I would have noticed it earlier if it were happening...

it happens pretty consistently, I've already spent coin 5 times since updating iOS and every time it shows my last number pressed for about 2 whole seconds.

**Comment by voisine** [comment]
@dabura667 is it only when spending, or every time you unlock? Can you check your most recent tx and tell me how many tx inputs it has? I wonder if it's pausing while signing the inputs. My test version is using libsecp256k1 which is significantly faster.

**Comment by dabura667** [comment]
1 input.

I have tried a few more times, and I have deduced it is only happening with payment requests. (both coinbase and bitpay) it does not happen with URIs with amounts, or when I manually type in an amount.

Maybe it has to do with cert verification?

I also think I might be kicking up the QR scanner, and typing in my PIN pretty fast immediately after opening the app for the first time in an hour or more... so it could be doing a lot all at once...

I'll try to control by unlocking, and waiting on the screen until I'm synced, read the payment request for a few seconds, and slowly type my PIN... maybe that will give it time.

On the plus side, testing this finally got me to buy Besiege, which is a really fun game.
Though, I wish I had a Mac and a developers account to build apis...
